pkg/flowaggregator/flowaggregator.go

```diff
@@ -324,28 +329,50 @@ func (fa *flowAggregator) initExportingProcess() error {
 		return fmt.Errorf("got error when initializing IPFIX exporting process: %v", err)
 	}
 	fa.exportingProcess = ep
-	// Currently, we send two templates for IPv4 and IPv6 regardless of the IP families supported by cluster
-	fa.templateIDv4 = fa.exportingProcess.NewTemplateID()
-	bytesSent, err := fa.sendTemplateSet(false)
+	// Currently, we send 4 templates for IPv4 and IPv6 regardless of the IP families supported by cluster
+	fa.templateIDv4Expv4 = fa.exportingProcess.NewTemplateID()
+	fa.templateIDv4Expv6 = fa.exportingProcess.NewTemplateID()
+	fa.templateIDv6Expv4 = fa.exportingProcess.NewTemplateID()
+	fa.templateIDv6Expv6 = fa.exportingProcess.NewTemplateID()
+	bytesSent, err := fa.sendTemplateSet(false, false)
+	if err != nil {
+		fa.exportingProcess.CloseConnToCollector()
+		fa.exportingProcess = nil
+		fa.templateIDv4Expv4 = 0
+		fa.set.ResetSet()
+		return fmt.Errorf("sending IPv4 template set with IPv4 original exporter ip failed, err: %v", err)
+	}
+	klog.V(2).Infof("Initialized exporting process and sent %d bytes size of IPv4 template set with IPv4 original exporter ip", bytesSent)
+
```

Comment:
Have a function that covers all four cases to avoid repetition?

Reply:
Done. Thanks